### PR TITLE
Resolving the null pointer error in the latest O3 snapshot setup

### PR DESCRIPTION
**Pull request by wikumChamith**
## Description of what I changed

In distro properties after 3.0.0-beta.16 frontend properties have been moved to the spa-assemble-config.json. This provides support for that.

## Issue I worked on

see https://issues.openmrs.org/browse/SDK-

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean install` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute the above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by dkayiwa** [comment]
@wikumChamith did you test this all the way through up to setting up a new O3 backend server which fully works up to being able to log in?

**Comment by wikumChamith** [comment]
Yes, I tested this.

**Comment by dkayiwa** [comment]
> Yes, I tested this.

Do you mind trying once more? It does not seem to work for me. It throws some errors on the server side log.

**Comment by wikumChamith** [comment]
@dkayiwa It works fine for me. Any errors I noticed are common errors we get when doing the initial setup for O3.
Here are my logs: https://gist.github.com/wikumChamith/f679d4f1c01aa44fc6f06c718dbf4987

File: maven-plugin/src/main/java/org/openmrs/maven/plugins/Setup.java [hunk 1]
```diff
@@ -211,8 +211,13 @@ private DistroProperties resolveDistroProperties(Server server) throws MojoExecu
 					Properties frontendProperties;
 					if (server.getVersion().equals(versionsHelper.getLatestSnapshotVersion(artifact))) {
 						frontendProperties = PropertiesUtils.getFrontendPropertiesFromSpaConfigUrl(
-								"https://raw.githubusercontent.com/openmrs/openmrs-distro-referenceapplication/main/frontend/spa-build-config.json");
-					} else {
+								"https://raw.githubusercontent.com/openmrs/openmrs-distro-referenceapplication/main/frontend/spa-assemble-config.json");
+					}
+					else if (new Version(server.getVersion()).higher(new Version("3.0.0-beta.16"))) {
```
Comment (dkayiwa) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/Setup.java:216]: Do you mind explaining this a bit? For i see both files at: https://github.com/openmrs/openmrs-distro-referenceapplication/tree/main/frontend

Comment (wikumChamith) [reply maven-plugin/src/main/java/org/openmrs/maven/plugins/Setup.java:216]: So here we fetch the frontend modules and add them to the distro.properties. Looks like in newer version those modules were moved to the spa-assemble-config.json file. 

https://github.com/openmrs/openmrs-distro-referenceapplication/commit/158db6182f1b652012755656a1f563ca1d34a900

**Pull request merged**